framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3094,15 +3095,35 @@ protected void checkThrownExpression(ThrowTree tree) {
         AnnotatedTypeMirror throwType = atypeFactory.getAnnotatedType(tree.getExpression());
         TypeMirror throwTM = throwType.getUnderlyingType();
         Set<? extends AnnotationMirror> required = getThrowUpperBoundAnnotations();
+        if (methodTree != null && getExceptionList(methodTree) != null) {
+            List<AnnotatedTypeMirror> exceptionList = getExceptionList(methodTree);
+            for (AnnotatedTypeMirror exception : exceptionList) {
+                Types typesUtil = atypeFactory.getProcessingEnv().getTypeUtils();
+                if (typesUtil.isSubtype(
+                        exception.getUnderlyingType(), throwType.getUnderlyingType())) {
+                    required = exception.getEffectiveAnnotations();
+                    break;
+                }
+            }
+        }
         switch (throwType.getKind()) {
             case NULL:
             case DECLARED:
+                AnnotationMirrorSet found = throwType.getAnnotations();
```

Comment:
Why does this suddenly need very similar code? Why doesn't the comparison in the TV/WC case below work?

Reply:
> Why does this suddenly need very similar code? Why doesn't the comparison in the TV/WC case below work?

Thanks, these code are duplicate and should be removed.